tests/lib/rules/jsx-key.js

```diff
@@ -68,6 +68,9 @@ ruleTester.run('jsx-key', rule, {
   }, {
     code: '[1, 2 ,3].map(x => { return <App /> });',
     errors: [{message: 'Missing "key" prop for element in iterator'}]
+  }, {
+    code: '[1, 2, 3]?.map(x => <App />)',
+    errors: [{message: 'Missing "key" prop for element in iterator'}]
```

Comment:
this will have to be restricted to the babel and `@typescript` parsers, until eslint supports it.

Reply:
Yeah, that's right